src/users/user.php

```diff
@@ -0,0 +1,367 @@
+<?php
+/**
+ * @package     Com.Api
+ * @subpackage  users
+ * @copyright   Copyright (C) 2009-2017 Techjoomla, Techjoomla Pvt. Ltd. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+// No direct access.
+defined('_JEXEC') or die();
+
+/**
+ * User Api.
+ * Creates a new user, updates an existing user and gets data of an user
+ *
+ * @package  Com.Api
+ *
+ * @since    2.0
+ */
+class UsersApiResourceUser extends ApiResource
+{
+	/**
+	 * Function to create and edit user record.
+	 *
+	 * @return object|void User details on success. raise error on failure.
+	 *
+	 * @since   2.0
+	 */
+	public function post()
+	{
+		$app = JFactory::getApplication();
+		$userIdentifier = $app->input->get('id', 0, 'String');
+		$formData = array();
+		$formData['username'] = $app->input->get('username', 0, 'String');
+		$formData['name'] = $app->input->get('name', 0, 'String');
+		$formData['email'] = $app->input->get('email', 0, 'String');
+		$formData['enabled'] = $app->input->get('enabled', 0, 'int');
+		$formData['activation'] = $app->input->get('activation', 0, 'int');
+		$formData['password'] = $app->input->get('password', 0, 'String');
+		$formData['groups'] = $app->input->get('groups', 0, 'Array');
+
+		$params = JComponentHelper::getParams("com_users");
+		$response = new stdClass;
+
+		$xidentifier = $app->input->server->get('HTTP_IDENTIFIER');
+		$fidentifier = $app->input->server->get('HTTP_FOURCECREATE');
+
+		if ($formData['username'] == '' || $formData['name'] == '' || $formData['email'] == '')
+		{
+			ApiError::raiseError(400, JText::_('PLG_API_USERS_REQUIRED_DATA_EMPTY_MESSAGE'));
+
+			return;
+		}
+
+		// Get current logged in user.
+		$my = JFactory::getUser();
+
+		// Check if $userIdentifier is not set
+		if (empty($userIdentifier))
+		{
+			if ($formData['password'] == '')
+			{
+				ApiError::raiseError(400, JText::_('PLG_API_USERS_REQUIRED_DATA_EMPTY_MESSAGE'));
+
+				return;
+			}
+
+			// Set default group if nothing is passed for group.
+			if (empty($formData['groups']))
+			{
+				$formData['groups'] = array($params->get("new_usertype", 2));
+			}
+
+			// Get a blank user object
+			$user = new JUser;
+
+			// Create new user.
+			$response = $this->storeUser($user, $formData, 1);
```

Comment:
Where is the ACL check for saving new user ?

Reply:
I don't think we have ACL check for registration in Joomla